src/third_party/dartino/mark_sweep.h

```diff
@@ -79,30 +79,26 @@ class FreeList {
 #endif
 
   void AddChunk(uword free_start, uword free_size) {
-    // If the chunk is too small to be turned into an actual
-    // free list chunk we turn it into fillers to be coalesced
-    // with other free chunks later.
-    if (free_size < FreeListChunk::SIZE) {
-      ASSERT(free_size <= 2 * WORD_SIZE);
-      Object** free_address = reinterpret_cast<Object**>(free_start);
-      for (uword i = 0; i * WORD_SIZE < free_size; i++) {
-        free_address[i] = StaticClassStructures::one_word_filler_class();
-      }
+    FreeListChunk* result = FreeListChunk::create_at(free_start, free_size);
+    if (!result) {
+      // If the chunk is too small to be turned into an actual
```

Comment:
Is the comment still accurate?

Reply:
Pretty much, but I edited it for clarity.